### DISCUSSION TITLE
swr interceptor and instance support

### DISCUSSION THREAD

**Discussion by muratdogan17** [vercel/swr#489] [Q&A]
Hi folks, this is really awesome library. I need two feature to replace my existing Axios implementation but couldn't find in any document, issue or pull request.

1 - I need to create several instances which have different base URLs
2 - I also need interceptors for both request and responses

Regards,
Murat

**Comment by sergiodxa** [comment]
I think both are possible right now, you can create as many instances as you want and use them as fetcher, same for interceptors. SWR is not tied to Axios and should not have features only to use them with this library.

**Reply by muratdogan17** [reply]
Hi Sergio, is it possible to share some code samples for both cases. I'm not familiar with SWR, that's why it is quite hard for me to get the point.

**Reply by sergiodxa** [reply]
I don't use Axios so I don't really know how to use interceptors and instances, but I will create a fake example:

Let's say we create two instances passing a base URL to pick the version of the API, and for one of them I add a custom interceptor:

```js
const instanceV1 = getAxiosInstance("/api/v1");
const instanceV2 = getAxiosInstance("/api/v2");

instanceV1.setInterceptor(customInterceptor);
```

You can then use them in SWR as your fetchers like this:

```js
const { data: users } = useSWR("users", instanceV1); // get users from v1 of the API
const { data: comments } = useSWR("comments", instanceV2); // get comments from v2 of the API
```

This way you can have any amount of instances with their interceptors, you will then pass those instances as second argument of SWR and then SWR will use it.

**Comment by akira-toriyama** [comment]
@muratdogan17 

https://github.com/bird-studio/use-map

You can use it if you like.
